pkg/render/guardian.go

```diff
@@ -277,3 +282,51 @@ func (c *GuardianComponent) container() []v1.Container {
 		},
 	}
 }
+
+// This is a service account that is being impersonated when calls are made from the management cluster to do
+// subject access reviews in a managed cluster.
+func (c *GuardianComponent) managerImpersonationServiceAccount() *v1.ServiceAccount {
```

Comment:
Instead of creating a second function here for the SA, Role, and Binding, WDYT about making common ones from the ones in `pkg/render/manager.go` and pass in a flag to set which version to use? I think only the ClusterRole would change so that is probably the only one that needs the flag.
I'm thinking I don't want to see one of these get changed and not the other.
It looks like there is no reason for the function to be attached to the component.

Reply:
I didn't really want to break a pattern of them being methods, but seeing that you suggest it, I'm on board with the idea.